src/invert/laplacexy/laplacexy.cxx

```diff
@@ -1713,73 +1717,60 @@ void LaplaceXY::solveFiniteDifference(const Field2D& x0) {
 }
 
 /*! Preconditioner
- * NOTE: For efficiency, this routine does not use globalIndex() 
- * in the inner loop. Instead, the indexing must be ordered in
- * exactly the same way as in the construction of indexXY
+ * NOTE: For generality, this routine does use globalIndex() in the inner loop, although
+ * this may be slightly less efficient than incrementing an integer for the global index,
+ * the finite-volume and finite-difference implementations have slightly different
+ * indexing patterns, so incrementing an integer would be tricky.
  */
 int LaplaceXY::precon(Vec input, Vec result) {
   
-  // Starting index
-  int ind = -1;
-  
   RangeIterator itdwn=localmesh->iterateBndryLowerY();
   if(!itdwn.isDone()) {
-    ind = globalIndex(itdwn.ind, localmesh->ystart-1);
     
     for(; !itdwn.isDone(); itdwn++) {
       // Should not go into corner cells, LaplaceXY stencil does not include them
       if (itdwn.ind < localmesh->xstart or itdwn.ind > localmesh->xend) {
         continue;
       }
+      const int ind = globalIndex(itdwn.ind, localmesh->ystart-1);
```

Comment:
With this change, can the loop be rewritten to 

```cpp
for (RangeIterator it = localmesh->iterateBndryLowerY(); !it.isDone(); it++) {
```

to match the other loops?